ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/transaction/arjunacore/BaseTransaction.java

```diff
@@ -308,6 +310,21 @@ public Void call() throws InvalidTransactionException,
 
 	private static final int _asyncCommitPoolSize = jtaPropertyManager.getJTAEnvironmentBean().getAsyncCommitPoolSize();
 
-	private static final ThreadPoolExecutor tpe = new ThreadPoolExecutor(1, _asyncCommitPoolSize, 10, TimeUnit.SECONDS, new ArrayBlockingQueue<Runnable>(3));
+	static class NamedThreadFactory implements ThreadFactory {
+		private final String name;
+		private final AtomicInteger counter = new AtomicInteger();
+
+		NamedThreadFactory(String name) {
+			this.name = name;
+		}
+
+		public Thread newThread(Runnable r) {
+			return new Thread(r, name + "-Thread_" + counter.incrementAndGet());
+		}
+	}
+
+	private static final ThreadFactory transactionThreadFactory = new NamedThreadFactory("Arjuna-Transaction");
```

Comment:
@mmusgrov - if we go ahead with this kind of change then I think maybe the threads would be prefixed with something like "Narayana" instead

Reply:
@tomjenkinson so you think the code needs to be:

```java
new NamedThreadFactory("Narayana");
```

Reply:
I think if we want this general type of behaviour from this PR then specifically in this are then it could be more appropriate to use "Narayana" than "Arjuna" in the wording. I then guess "Narayana" rather than "Narayana-Transaction" - if the "-Transaction" suffix can help that could be considered.

Reply:
@krosenvold can you please use as the prefix just word `Narayana`?